database/index.js

```diff
@@ -1,37 +1,68 @@
 const mongoose = require('mongoose');
+
 mongoose.connect('mongodb://root:WissemGamra1@ds211143.mlab.com:11143/artists');
 
 const db = mongoose.connection;
-
+db.on('error', () => console.log('mongoDB connection error'));
+db.once('open', () => console.log('mongoDB connection established'));
 
 const ArtistSchema = new mongoose.Schema({
   artistID: Number,
   artistName: String,
-  albums: [{
-    albumID: Number,
-    albumName: String,
-    albumImage: String,
-    publishedYear: Number,
-    songs: [{
-      songID: Number,
-      songName: String,
-      streams: Number,
-      length: Number,
-      popularity: Number,
-      addedToLibrary: Boolean
-    }]
-  }]
+  albums: [
+    {
+      albumID: Number,
+      albumName: String,
+      albumImage: String,
+      publishedYear: Number,
+      songs: [
+        {
+          songID: Number,
+          songName: String,
+          streams: Number,
+          length: Number,
+          popularity: Number,
+          addedToLibrary: Boolean,
+        },
+      ],
+    },
+  ],
 });
 
-var Artist = mongoose.model('Artist', ArtistSchema);
+const Artist = mongoose.model('Artist', ArtistSchema);
 
-var getArtist = (id, cb) => {
-  Artist.find({'artistID': id}, (err, data) => {
-    if (err) throw err;
-    cb(data);
-  })
-}
+const getArtist = id => new Promise((res, rej) => {
+  Artist.find({ artistID: id }, (err, data) => {
+    if (err) rej(err);
+    res(data);
+  });
+});
+
+const insertArtists = artists => new Promise((res, rej) => {
+  Artist.create([...artists], (err, result) => {
+    if (err) rej(err);
+    res(result);
+  });
+});
+
+const updateArtist = (id, newValues) => new Promise((res, rej) => {
+  Artist.updateOne({ artistID: id }, { $set: newValues }, (err, results) => {
+    if (err) rej(err);
+    res(results);
+  });
+});
+
+const deleteArtist = id => new Promise((res, rej) => {
+  Artist.deleteOne({ artistID: id }, (err, results) => {
+    if (err) rej(err);
+    res(results);
+  });
+});
```

Comment:
This looks good, I would just add the response status for each request as best practice